### ISSUE TITLE
Players are swapping team colours

### ISSUE THREAD

**Issue by freezurbern** [red-eclipse/base#269]
![20150712092024](https://cloud.githubusercontent.com/assets/6363417/8637942/639a82a2-2878-11e5-8b46-8bd8cdb65431.png)
![20150712092026](https://cloud.githubusercontent.com/assets/6363417/8637944/639f9544-2878-11e5-8b73-836086d1d62f.png)
![20150712092217](https://cloud.githubusercontent.com/assets/6363417/8637943/639f5c8c-2878-11e5-8626-afc227e5ed52.png)
![20150712092218](https://cloud.githubusercontent.com/assets/6363417/8637945/63a0653c-2878-11e5-83f0-91bb8c4d35e4.png)

Demo of it happening: https://dl.dropboxusercontent.com/u/42922158/20150712092621.dmo

```
[09:20:56] <freezurbern> i found a really big bug
[09:21:05] <freezurbern> players are swapping team colors :(
[09:24:37] * ChanServ gives voice to re-master
[09:24:51] <Iceflower> lol yes
[09:27:55] <angeles> freezurbern: if it is still happening in the current version then it is something eihrul needs to address
[09:28:08] <angeles> I've been trying for ages to deal with it and I give up
[09:28:21] <freezurbern> oh so you know about it
[09:28:55] <angeles> yeah it's why I did those light tweaks, was just a collection of stuff I noticed while trying to debug it
[09:29:12] <freezurbern> it is VERY difficult to play bomberball with that bug :P
[09:29:26] <angeles> lulz
[09:29:45] <freezurbern> "no dont score on yourself!!" "enemy scored" "ARGH!!!!!"
[09:29:48] <angeles> it doesn't seem to be random, I think they're inheriting someone else's colour
[09:30:19] <freezurbern> the vanities are becoming colored differently lol
[09:30:22] <angeles> because it's not like it flashes colours either, they change colour for a while, then change back
```

Occured for me with server 638d5a5 and client ea076bb

**Comment by IceflowRE** [comment]
Just want to add, that not only players are changing the colour, flags too.

**Comment by freezurbern** [comment]
Seems to be fixed by d6314e92b49dbb0afb0b4778a17ffeccfa279562, Thanks! Hopefully this will be pushed to stable SOON
